modules/redis/options.go

```diff
@@ -0,0 +1,136 @@
+package redis
+
+import (
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"github.com/mdelapenya/tlscert"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+type options struct {
+	tlsEnabled bool
+	tlsConfig  *tls.Config
+}
+
+// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
+var _ testcontainers.ContainerCustomizer = (Option)(nil)
+
+// Option is an option for the Redpanda container.
+type Option func(*options) error
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// WithTLS sets the TLS configuration for the redis container, setting
+// the 6380/tcp port to listen on for TLS connections and using a secure URL (rediss://).
+func WithTLS() Option {
+	return func(o *options) error {
+		o.tlsEnabled = true
+		return nil
+	}
+}
+
+// WithConfigFile sets the config file to be used for the redis container, and sets the command to run the redis server
+// using the passed config file
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	const defaultConfigFile = "/usr/local/redis.conf"
```

Comment:
suggestion: either raise to file level to increase visibility and reuse opportunities or just use literal below.
Raising it to file level is consistent with other options.

Reply:
let me change this in a follow-up refactor, as it's not related to TLS. I moved the options to its own file that's why it appears as new code